query/src/storages/s3/s3_stage_source.rs

```diff
@@ -149,19 +151,24 @@ impl StageSource {
 
         let op = Self::get_op(&self.ctx, &self.table_info.stage_info).await?;
         let path = file_name.unwrap_or_else(|| "".to_string());
-        let file_reader = op.object(&path).reader();
+        let object = op.object(&path);
 
         // Get the format(CSV, Parquet) source stream.
         let source = match &file_format {
-            StageFileFormatType::Csv => {
-                Ok(Self::csv_source(ctx.clone(), self.schema.clone(), stage, file_reader).await?)
-            }
-            StageFileFormatType::Parquet => {
-                Ok(
-                    Self::parquet_source(ctx.clone(), self.schema.clone(), stage, file_reader)
-                        .await?,
-                )
-            }
+            StageFileFormatType::Csv => Ok(Self::csv_source(
+                ctx.clone(),
+                self.schema.clone(),
+                stage,
+                Box::new(object.reader().await?),
+            )
+            .await?),
+            StageFileFormatType::Parquet => Ok(Self::parquet_source(
+                ctx.clone(),
+                self.schema.clone(),
+                stage,
+                seekable_read(&object, ..),
```

Comment:
Here is not friendly to use, why not `object.seek_reader()`?

Reply:
I want to indicate that it's not a zero-cost operation. But I think it's better to have a `object.seekable_reader()`, let's do this.